include/knowhere/index_node.h

```diff
@@ -115,6 +115,72 @@ class IndexNode : public Object {
     Version version_;
 };
 
+// An iterator implementation that accepts a list of distances and ids and returns them in order.
+class PrecomputedDistanceIterator : public IndexNode::iterator {
+ public:
+    PrecomputedDistanceIterator(std::vector<std::pair<float, int64_t>>&& distances_ids, bool larger_is_closer)
+        : comp_(larger_is_closer), results_(std::move(distances_ids)) {
+        sort_size_ = std::max((size_t)50000, results_.size() / 10);
```

Comment:
would it be possible to move `std::max((size_t)50000, results_.size() / 10)` into a separate static function, bcz the same code here and below is identical?

Reply:
Done